pandas/core/algorithms.py

```diff
@@ -547,10 +566,12 @@ def mode(values):
         constructor = Series
 
     dtype = values.dtype
-    if is_integer_dtype(values):
+    if is_signed_integer_dtype(values):
         values = _ensure_int64(values)
         result = constructor(sorted(htable.mode_int64(values)), dtype=dtype)
```

Comment:
And how is that any different from `np.sort`, which converts to `ndarray` before we then convert to `Series`?  We might as well just use `Series.sort_values` then?